chall01/tdawson.py

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/python3
+import sys
+
+if len(sys.argv) != 2 or len(sys.argv[1]) == 0:
+    sys.exit(f'usage: {sys.argv[0]} <a-zA-Z string>')
+
+morse_alphabet = {'a': '.-', 'b': '-...', 'c': '-.-.', 'd': '-..', 'e': '.',
+                  'f': '..-.', 'g': '--.', 'h': '....', 'i': '..', 'j': '.---',
+                  'k': '-.-', 'l': '.-..', 'm': '--', 'n': '-.', 'o': '---', 'p': '.--.',
+                  'q': '--.-', 'r': '.-.', 's': '...', 't': '-', 'u': '..-', 'v': '...-',
+                  'w': '.--', 'x': '-..-', 'y': '-.--', 'z': '--..', ' ': ' '}
+
+string = sys.argv[1].lower()
+
+if not all(c in morse_alphabet for c in string):
+    sys.exit(f'usage: {sys.argv[0]} <a-zA-Z string>')
+
+for c in string:
+    print(morse_alphabet[c], end='')
```

Comment:
I would suggest avoiding printing char by char but better to append into a string and then print the string at the end.

Reply:
Hi, thank you for the feedback, it is very much appreciated! 👍 

I agree with your suggestion here, I learned about the `String.join()` method so I would now do it this way:
```suggestion
print(''.join(morse_alphabet[c] for c in string))
```